src/ex7/client/src/components/TodoList/TodoList.js

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getItemsList } from "../../selectors/items-entities-selectors";
+import LoaderTodos from "../LoaderTodos/LoaderTodos";
+import TodoItem from "../TodoItem/TodoItem";
+import TodoInput from "../TodoInput/TodoInput";
+import SearchTodo from "../SearchTodo/SearchTodo";
+import SortTodos from "../SortTodos/SortTodos";
+import filterTodosOptions from "../FilterTodosOptions/FilterTodosOptions";
+import "./TodoList.css";
+import {
+  loadtodosListAction,
+  removetodoAction,
+  updateItemAction,
+} from "../../actions/todo-actions";
+
+const TodoList = ({
+  todosList,
+  removetodoAction,
+  loadtodosListAction,
+  updateItemAction,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchInputValue, setSearchInputValue] = useState("");
+  const [filterStatusValue, setFilterStatusValue] = useState("");
+
+  const loadTodos = useCallback(async () => {
+    await loadtodosListAction();
+  }, [loadtodosListAction]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    loadTodos();
+    setIsLoading(false);
+  }, []);
+
+  const handleTodoDelete = useCallback(
+    async (todo) => {
+      removetodoAction(todo);
+    },
+    [removetodoAction]
+  );
+
+  const handleFilterTodos = useCallback((value) => {
+    setIsLoading(true);
+    setFilterStatusValue(value);
+    setIsLoading(false);
+  }, []);
+
+  const handleTodoUpdate = useCallback(
+    async (item) => {
+      updateItemAction(item);
+    },
+    [updateItemAction]
+  );
+
+  const handleTodoStatusUpdate = async (item) => {
+    item.status = !item.status;
+    await handleTodoUpdate(item);
+  };
+
+  const handleSearchTodo = useCallback((value) => {
+    setIsLoading(true);
+    setSearchInputValue(value);
+    setIsLoading(false);
+  }, []);
+
+  return (
+    <>
+      <TodoInput />
+      <SearchTodo
+        placeholder={"Search Todos"}
+        loading={isLoading}
+        onChange={handleSearchTodo}
+      />
+      <ul id="list">
+        {!isLoading ? (
+          todosList
+            .filter(
```

Comment:
consider:
- extract searchInputValue to the store with relevant actions
- extract to a selector